src/main/java/org/codehaus/mojo/versions/AbstractVersionsDependencyUpdaterMojo.java

```diff
@@ -334,11 +333,7 @@ private boolean compare( MavenProject project, Dependency dep )
         {
             return false;
         }
-        if ( !StringUtils.equals( project.getArtifactId(), dep.getArtifactId() ) )
-        {
-            return false;
-        }
-        return true;
+        return StringUtils.equals( project.getArtifactId(), dep.getArtifactId() );
```

Comment:
artifactId cannot be `null`, you can just use 
```suggestion
        return project.getArtifactId().equals( dep.getArtifactId() );
```